packages/cardpay-reward-programs/pyproject.toml

```diff
@@ -20,16 +20,19 @@ dependencies = [
     "pysha3>=1.0.2",
     "duckdb>=0.3.2",
     "python-dotenv>=0.19.2",
-    "future-fstrings>=1.2.0"]
-requires-python = ">=3.9"
+    "future-fstrings>=1.2.0",
+    "sentry-sdk>=1.5.12",
+]
+requires-python = ">=3.9, <3.11"
 license = {text = "MIT"}
 
 [project.urls]
 Homepage = ""
 [tool.pdm]
 [project.optional-dependencies]
 explore = [
-    "streamlit>=1.9.0"
+    "streamlit>=1.9.0",
+    "hexbytes>=0.2.2",
```

Comment:
```
 (new2) tintinthong@Justins-MacBook-Pro cardpay-reward-programs % pdm add -G explore hexbytes                                                                                      
 Adding packages to explore dependencies: hexbytes                                                                                                                                 
 ✖ 🔒 Lock failed                                                                                                                                                                  
 Unable to find a resolution because the following dependencies don't work on all Python versions defined by the project's `requires-python`: >=3.9                                
   python>=3.6,<3.11 (from <Candidate eth-typing 2.3.0 from https://pypi.org/simple/eth-typing/>)                                                                                  
   python>=3.6,<3.11 (from <Candidate eth-typing 2.3.0 from unknown>)                                                                                                              
 To fix this, you can change the value of `requires-python` in pyproject.toml.                                                                                                     
 See /var/folders/hv/qnr8mqkn4zxcgt1rg18sbbjw0000gn/T/pdm-lock-zdoozpdq.log for detailed debug log.                                                                                
 [ResolutionImpossible]: Unable to find a resolution 
```
There are some resolution errors when adding `hexbytes`. I resolved this by narrowing the `requires-python` range. Probably related [issue](https://github.com/ethereum/eth-typing/pull/29)

Reply:
Oh interesting. I guess if we set an upper bound to the python version this package supports to 3.10 this might go away?

Reply:
Yup this is what I did it seemed to work